presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySliceFunction.java

```diff
@@ -43,7 +43,7 @@ public static Block slice(
         checkCondition(fromIndex != 0, INVALID_FUNCTION_ARGUMENT, "SQL array indices start at 1");
 
         int size = array.getPositionCount();
-        if (size == 0) {
+        if (size == 0 || (fromIndex == 1 && size < fromIndex + length)) {
```

Comment:
Do we have unit test coverage for the new condition?

Reply:
Yes like

https://github.com/prestodb/presto/blob/3fc62731e0e955036160eba8ab16d3386802d139/presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java#L1108